src/rocq_elpi_utils.ml

```diff
@@ -763,6 +763,17 @@ let detype_closed_glob env sigma closure =
   fix_detype gbody
 [%%endif]
 
+[%%if coq = "8.19" || coq = "8.20" || coq = "9.0"]
+let detype_to_pattern env sigma c =
+  let c = detype env sigma c in
+  Patternops.pattern_of_glob_constr env c
+[%%else]
+let detype_to_pattern env sigma c =
+  let c = detype env sigma c in
+  let ids, p = Patternops.pattern_of_glob_constr env c in
```

Comment:
BTW is going through detyping better than pattern_of_constr

Reply:
I don't understand this sentence

Reply:
FTR, *my* detype is not really like the one of Rocq, if only does not depend on printing flags